scom/stackstate_checks/scom/scom.py

```diff
@@ -82,114 +52,136 @@ def sendAlerts(self, session, scom_ip, domain, username, password, component_id)
+        else:
+            return str(health)
+    def scom_powershell_check(self,scom_server):
+        components = self.excute_powershell_cmd (scom_server,"C:\\ProgramData\\StackState\\conf.d\\scom.d\\components.ps1")
+        for cmp in components:
+            properties_list = dict()
+            for key,value in cmp.items():
+                properties_list.update({str(key).encode('utf-8'):str(value).encode('utf-8')})
+            name = str(cmp["DisplayName"])
+            component_id =  str(cmp["Id"])
+            health = self.get_health_powershell(cmp["HealthState"])
+            component_type = str(cmp["FullName"].lower().split(":")[0])
+            properties_list.update({"id":str(component_id),"name":name})
+            self.component(component_id,component_type,properties_list)
             self.event({
                 "timestamp": int(time.time()),
                 "msg_title": "Health Status",
                 "msg_text": "Change in health status",
                 "source_type_name": "Health",
-                "host": idd.get("displayName"),
+                "host": name,
                 "tags": [
-                    "status:%s" % get_entity_health_state(session, scom_ip, domain, username, password,
-                                                          idd.get("healthIconUrl")),
-                    "id:%s" % idd.get("id")
+                    "status:%s" % health,
+                    "id:%s" % str(component_id)
                 ]
             })
-            self.relation(str(serviceid), str(idd.get("id")), "hosted_on", {})
-            self.sendAlerts(session, scom_ip, domain, username, password, idd.get("id"))
-        return ids
-
-    def serviceTree(self, serviceidtree, session, scom_ip, domain, username, password, system_type):
-        for serviceid in serviceidtree:
-            if not hasChild(serviceid, session, scom_ip, domain, username, password):
-                continue
-            else:
-                self.serviceTree(self.nextNodes(serviceid, session, scom_ip, domain, username, password, system_type),
-                                 session, scom_ip, domain, username, password, system_type)
+        relations = self.excute_powershell_cmd (scom_server,"C:\\ProgramData\\StackState\\conf.d\\scom.d\\relations.ps1")
+        for rel in relations:
+            source_id = str(rel["source"]["Id"])
+            target_id = str(rel["target"]["Id"])
+            self.relation(source_id,target_id,"Is_connected_to",{})
 
     def check(self, instance):
-        # read configuration file/yaml configuration
         scom_ip = str(instance.get('hostip'))
         domain = str(instance.get('domain'))
         username = str(instance.get('username'))
         password = str(instance.get('password'))
         auth_method = str(instance.get('auth_mode'))
+        integration_mode = str(instance.get('integration_mode'))
+        self.requests_threshold = instance.get('max_number_of_requests')
```

Comment:
Can you set defaults for these or check there existence before starting the check

Reply:
defaults added